slick/src/main/scala/slick/jdbc/JdbcBackend.scala

```diff
@@ -408,17 +410,11 @@ trait JdbcBackend extends RelationalBackend {
   }
 
   class BaseSession(val database: Database) extends SessionDef {
-    protected var open = false
     protected var inTransactionally = 0
 
-    def isOpen = open
     def isInTransaction = inTransactionally > 0
 
-    lazy val conn = {
-      val c = database.source.createConnection
-      open = true
-      c
-    }
+    val conn = database.source.createConnection
```

Comment:
Good, it's time we got rid of the unnecessary laziness that was there in 3.0 for backwards compatibility with the old API.